File: app/src/main/java/com/aol/philipphofer/gui/custom/CustomActivity.java
package com.aol.philipphofer.gui.custom;

import android.app.Activity;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.aol.philipphofer.R;
import com.aol.philipphofer.persistence.Data;

public abstract class CustomActivity extends AppCompatActivity{

    protected Data data;

    @Override
    protected void onCreate(Bundle savedBundle) {
        super.onCreate(savedBundle);
        data = Data.instance(this);

        System.out.println(getTheme());
    }
}

File: app/src/main/java/com/aol/philipphofer/gui/custom/CustomActivity.kt
package com.aol.philipphofer.gui.custom

import android.app.Activity
import android.os.Bundle
import com.aol.philipphofer.persistence.Data

abstract class CustomActivity : Activity() {

    protected lateinit var data: Data

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        data = Data.instance(this);
    }
}